drop-transfer/src/ws/server/v6.rs

```diff
@@ -580,8 +585,38 @@ impl handler::Downloader for Downloader {
                 .join(super::temp_file_name(task.xfer.id(), task.file.id())),
         );
 
+        let tmp_size = fs::File::open(&tmp_location.0)
+            .and_then(|file| file.metadata())
+            .map(|metadata| metadata.len())
+            .ok();
+
+        let will_emit_checksum_events = emit_checksum_events && tmp_size.is_some();
+
+        let cb = if will_emit_checksum_events {
+            let size = tmp_size.unwrap_or(0);
+            events.checksum_start(size).await;
+
+            Some({
+                let ev = events.clone();
+                move |progress: u64| {
+                    let ev = ev.clone();
+                    async move {
+                        ev.checksum_progress(progress).await;
+                    }
+                }
+            })
+        } else {
+            None
+        };
+
+        let res = super::TmpFileState::load(&tmp_location.0, cb).await;
+
+        if will_emit_checksum_events {
+            events.checksum_finish().await;
+        }
+
```

Comment:
TBH, I somewhat have a problem with this logic here. It's shared among two versions and clutters the `Download` trait significantly.
I suggest expanding the `DownloadInit` enum into two cases: resuming and starting from the beginning and offloading the logic higher in the call stack

Reply:
I've gained some simplification by reducing changes to v4 and v6 files by moving the logic one layer above